src/components/Ideas/CompIdeaList.js

```diff
@@ -0,0 +1,42 @@
+import React, { useContext, useEffect, useState } from "react"
+import { useHistory } from "react-router-dom"
+import { IdeaContext } from "./IdeaProvider"
+import { CompIdeaDetail } from "./CompIdeaDetail"
+import "./Idea.css"
+
+export const CompIdeaList = () => {
+    const { ideas, getCompleteIdeas, searchTerms } = useContext(IdeaContext)
+    const [ filteredIdeas, setFiltered ] = useState([])
+    const history = useHistory()
+
+    useEffect(() => {getCompleteIdeas()}, [])
+
+    useEffect(() => {
+        if (searchTerms !== "") {
+            const subset = ideas.filter(idea => (
+                idea.desc.toLowerCase().includes(searchTerms.toLowerCase()) ||
```

Comment:
Id store `searchTerms.toLowerCase()` in a variable outside the function because you're using it twice. For performance, this method will have to run each time it's called which makes the browser work twice as hard.